source/change-streams/tests/change-streams.yml

```diff
@@ -0,0 +1,197 @@
+collection_name: &collection_name "test"
+database_name: &database_name "change-stream-tests"
+collection_name_2: &collection_name_2 "test2"
+database_name_2: &database_name_2 "change-stream-tests-2"
+defaults:
+  minServerVersion: "3.6.0"
+  topology: replicaset
+  target: collection
+  numChanges: 1
+  result: {}
+  expectations: []
+  operations:
+    -
+      database: *database_name
+      collection: *collection_name_2
+      commandName: insert
+      arguments:
+        - { x: 1 }
+    -
+      database: *database_name_2
+      collection: *collection_name
+      commandName: insert
+      arguments:
+        - { y: 2 }
+    -
+      database: *database_name
+      collection: *collection_name
+      commandName: insert
+      arguments:
+        - { z: 3 }
+tests:
+  -
+    description: "`$changeStream` must be the first stage in a change stream pipeline sent to the server"
+    expectations:
+      - 
+        command_started_event:
+          command:
+            aggregate: *collection_name
+            pipeline:
+              - {$changeStream: {}}
+          command_name: aggregate
+          database_name: *database_name
+  -
+    description: The watch helper must not throw a custom exception when executed against a single server topology, but instead depend on a server error
+    topology: single
+    operations: []
+    result:
+      error:
+        errmsg: The $changeStream stage is only supported on replica sets
+        code: 40573
+  -
+    description: ChangeStream must continuously track the last seen resumeToken
```

Comment:
Does this require a failpoint?

Reply:
This one I wasn't sure of. I mostly copy-pasted the current prose tests in here. The issue with this one is that we'd have to introduce an entirely new type of assertion to cover this use case, and I was not sure of the best way to go about that. How does each driver track the `resumeToken`, and what is the best way to express an assertion based on `resumeToken`?

Reply:
Thinking about it a bit more I think it will require some sort of failpoint.  Because the point is that on a failure the driver resume with the proper resumeToken, and we need a failpoint to force a failure that requires resuming

Reply:
done: removed all skippable tests